packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -20,14 +27,278 @@ import {
   DROPDOWN_CHIP_COLORS,
   DROPDOWN_ID,
   DROPDOWN_MENU_PLACEMENT,
-  DROPDOWN_MENU_POSITION
+  DROPDOWN_MENU_POSITION,
+  DropdownOption
 } from "./DropdownConstants";
 import generateBaseStyles, { customTheme } from "./Dropdown.styles";
 import Control from "./components/Control/Control";
 import menuStyles from "./components/menu/menu.module.scss";
 import styles from "./Dropdown.module.scss";
+import { VibeComponent, VibeComponentProps } from "../../types";
 
-const Dropdown = forwardRef(
+interface CustomSingleValueProps extends SingleValueProps<DropdownOption> {
+  Renderer: React.ComponentType;
+  readOnly: boolean;
+  selectedOption: DropdownOption;
+}
+
+interface CustomMenuProps extends MenuProps<OptionTypeBase, boolean> {
+  dropdownMenuWrapperClassName: string;
+}
+
+interface CustomOptionProps extends OptionProps<OptionTypeBase, boolean> {
+  optionWrapperClassName?: string;
+}
+
+type SelectEvent = {
+  action: "select-option" | "clear";
+  option?: DropdownOption;
+};
+
+type DropdownState = {
+  isDisabled: boolean;
+  selectProps: { withReadOnlyStyle: boolean; readOnly: boolean };
+};
+
+export interface DropdownComponentProps extends VibeComponentProps {
+  /**
+   * Custom style
+   */
+  className?: string;
+  /** ClassName to be added to dropdown option wrapper (dropdown-wrapper__option--reset) */
+  optionWrapperClassName?: string;
+  /** ClassName to be added to dropdown single value wrapper (dropdown-wrapper__single-value--reset) */
+  singleValueWrapperClassName?: string;
+  /** ClassName to be added to dropdown menu wrapper (dropdown-menu-wrapper) */
+  dropdownMenuWrapperClassName?: string;
+  /**
+   * Placeholder to show when no value was selected
+   */
+  placeholder?: string;
+  /**
+   * If set to true, dropdown will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * If set to true, dropdown won't be editable
+   */
+  readOnly?: boolean;
+  /**
+   * Called when menu is opened
+   */
+  onMenuOpen?: (event: React.MouseEvent) => void;
+  /**
+   * Called when menu is closed
+   */
+  onMenuClose?: (event: React.MouseEvent) => void;
+  /**
+   * Called when key is pressed in the dropdown
+   */
+  onKeyDown?: (event: React.MouseEvent) => void;
+  /**
+   * Called when focused
+   */
+  onFocus?: (event: React.MouseEvent) => void;
```

Comment:
```suggestion
  onFocus?: FocusEventHandler
```

`FocusEventHandler` is imported from `react-select` alongside the other types

the `onFocus` is not a `MouseEvent`